### PR TITLE
Implemented: Filter jobs based upon pinned jobs(#2cqzpye)

### PR DESCRIPTION
**Pull request by rathoreprashant**
### Related Issues
Filter jobs based upon pinned jobs (#2cqzpye)

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/views/Pipeline.vue [hunk 1]
```diff
@@ -359,7 +361,23 @@ export default defineComponent({
       getPinnedJobs: 'user/getPinnedJobs'
     })
   },
-  methods: {
+  methods : {
+    isSelected(JobEnumId: any) {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:365]:
```suggestion
    isPinnedJobSelected(JobEnumId: any) {
```

File: src/views/Pipeline.vue [hunk 2]
```diff
@@ -359,7 +361,23 @@ export default defineComponent({
...
+    return (this as any).fEnumId.some((jobId: any) =>  jobId === JobEnumId );  
+    },
+    updateFilteredEnum(JobEnumId: any) {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:368]:
```suggestion
    updateSelectedPinnedJob(JobEnumId: any) {
```

File: src/views/Pipeline.vue [hunk 3]
```diff
@@ -326,6 +326,8 @@ export default defineComponent({
   },
   data() {
     return {
+      fEnumId:[],
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:329]:
```suggestion
      selectedPinnedJobs:[],
```

File: src/views/Pipeline.vue [hunk 4]
```diff
@@ -359,7 +361,23 @@ export default defineComponent({
...
     })
   },
-  methods: {
+  methods : {
+    isSelected(JobEnumId: any) {
+    return (this as any).fEnumId.some((jobId: any) =>  jobId === JobEnumId );  
+    },
+    updateFilteredEnum(JobEnumId: any) {
+      
+      const isAlreadyAdded = this.isSelected(JobEnumId);
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:370]: Get indexOf, use it to check if it is already in list and then perform operation

File: src/views/Pipeline.vue [hunk 5]
```diff
@@ -535,6 +553,8 @@ export default defineComponent({
       } else {
         pinnedJobs.add(enumId);
       }
+      
+      this.updateFilteredEnum(enumId)
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:557]: I think this should be done after pinned jobs is updated

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 4 above)
Comment (rathoreprashant) [reply src/views/Pipeline.vue]: Improved sir

File: src/views/Pipeline.vue [hunk 6]
```diff
@@ -359,7 +360,22 @@ export default defineComponent({
       getPinnedJobs: 'user/getPinnedJobs'
     })
   },
-  methods: {
+  methods : {
+    isPinnedJobSelected(jobEnumId: any) {
+    return (this as any).selectedPinnedJobs.some((jobId: any) =>  jobId === jobEnumId );
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:365]:
```suggestion
      return (this as any).selectedPinnedJobs.some((jobId: any) =>  jobId === jobEnumId );
```

**Pull request merged**